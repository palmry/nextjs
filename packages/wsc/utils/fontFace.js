import { css } from "styled-components"

export const createFontStyles = (styles) => {
  let fontStyles = ``
  styles.forEach((style) => {
    fontStyles += `
    @font-face {
      font-family: '${style.fontFamily}';
      src: url(${style.url}) format('${style.format}');
      font-weight: ${style.fontWeight};
      font-style: ${style.fontStyle};
      font-display: swap;
    }
    `
  })
  return css`
    ${fontStyles}
  `
}
