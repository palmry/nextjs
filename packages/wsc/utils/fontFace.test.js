import { createFontStyles } from './fontFace'

describe('fontFace', () => {
  it('should create font styles', () => {
    const AVENIR_FONT_CONFIG = [
      {
        fontFamily: 'Avenir Next LT Pro',
        url: 'AvenirNextLTPro-Bold.otf',
        format: 'opentype',
        fontWeight: 'bold',
        fontStyle: 'normal',
      },
      {
        fontFamily: 'Avenir Next LT Pro',
        url: 'AvenirNextLTPro-It.otf',
        format: 'opentype',
        fontWeight: 'bold',
        fontStyle: 'italic',
      },
    ]

    const css = createFontStyles(AVENIR_FONT_CONFIG)
    expect(css).toMatchSnapshot()
  })
})
