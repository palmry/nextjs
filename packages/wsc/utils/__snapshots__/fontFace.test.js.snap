// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontFace should create font styles 1`] = `
Array [
  "
    ",
  "
    @font-face {
      font-family: 'Avenir Next LT Pro';
      src: url(AvenirNextLTPro-Bold.otf) format('opentype');
      font-weight: bold;
      font-style: normal;
      font-display: swap;
    }
    
    @font-face {
      font-family: 'Avenir Next LT Pro';
      src: url(AvenirNextLTPro-It.otf) format('opentype');
      font-weight: bold;
      font-style: italic;
      font-display: swap;
    }
    ",
  "
  ",
]
`;
